helix-core/src/main/java/org/apache/helix/common/caches/CustomizedViewCache.java

```diff
@@ -0,0 +1,146 @@
+package org.apache.helix.common.caches;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.helix.HelixDataAccessor;
+import org.apache.helix.HelixException;
+import org.apache.helix.PropertyKey;
+import org.apache.helix.PropertyType;
+import org.apache.helix.model.CustomizedView;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Maps;
+
+/**
+ * Cache to hold all CustomizedView of a specific type.
+ */
+public class CustomizedViewCache extends AbstractDataCache<CustomizedView> {
+  private static final Logger LOG = LoggerFactory.getLogger(CustomizedViewCache.class.getName());
+
+  protected Map<String, CustomizedView> _customizedViewMap;
+  protected Map<String, CustomizedView> _customizedViewCache;
+  protected String _clusterName;
+  private PropertyType _propertyType;
+  private String _type;
```

Comment:
Please see above comment. "type" is very unclear.

Reply:
fixed.